app/validators/questionnaire_schema.py

```diff
@@ -473,10 +478,9 @@ def _get_answer_maximum(
     def _get_numeric_value(self, defined_value, system_default, answer_ranges):
         if not isinstance(defined_value, dict):
             return defined_value
-        if source := defined_value.get("source"):
+        if defined_value.get("source"):
             referred_answer = self.get_numeric_value_for_value_source(
-                value_source=source,
-                defined_value=defined_value,
+                value_source=defined_value,
```

Comment:
Just a detail. Positions are not forced in that method and there is only two of them so just:
```python
self.get_numeric_value_for_value_source(defined_value, answer_ranges)
```